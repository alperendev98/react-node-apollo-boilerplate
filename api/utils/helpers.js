function getValidationRules(allowGraphqlIntrospection, graphqlDisableIntrospection) {
	if (!allowGraphqlIntrospection && !graphqlDisableIntrospection) return [];

	return allowGraphqlIntrospection ? [] : [graphqlDisableIntrospection];
}

module.exports = {
	getValidationRules
};
