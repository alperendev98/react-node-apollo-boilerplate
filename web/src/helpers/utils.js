import pickBy from 'lodash/pickBy';
import isObject from 'lodash/isObject';
import isArray from 'lodash/isArray';

export function processModifiers(modifiers) {
	return isObject(modifiers) && !isArray(modifiers) ? pickBy(modifiers) : modifiers;
}
