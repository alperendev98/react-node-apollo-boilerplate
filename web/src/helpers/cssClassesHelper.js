import BEMHelper from 'react-bem-helper';
import { processModifiers } from './utils';

export const getBEMClasses = (...classes) => {
	classes = Array.prototype.concat(...classes).filter(item => item);

	const BEMClasses = classes.map(
		className =>
			new BEMHelper({
				name: className,
				outputIsString: true
			})
	);
	return (element, modifier) =>
		BEMClasses.map(BEMClass => BEMClass(element, processModifiers(modifier))).join(' ');
};

export const addModifier = (modifiers, newModifier) => {
	let result;

	if (Array.isArray(modifiers)) {
		result = [newModifier, ...modifiers];
	} else if (typeof modifiers === 'string') {
		result = [newModifier, modifiers];
	} else if (typeof modifiers === 'object') {
		result = modifiers;
		result[newModifier] = true;
	}

	return result;
};
