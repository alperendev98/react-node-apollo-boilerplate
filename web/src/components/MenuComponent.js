import React from 'react';
import PropTypes from 'prop-types';
import Button from '@material-ui/core/Button';
import ClickAwayListener from '@material-ui/core/ClickAwayListener';
import Grow from '@material-ui/core/Grow';
import Paper from '@material-ui/core/Paper';
import Popper from '@material-ui/core/Popper';
import MenuItem from '@material-ui/core/MenuItem';
import MenuList from '@material-ui/core/MenuList';

import scheduleIcon from '../assets/images/scheduler.svg';
import refIcon from '../assets/images/ref.svg';
import conferencIcon from '../assets/images/conf.svg';
import flagIcon from '../assets/images/flag.svg';
import lcfIcon from '../assets/images/lcf.svg';

import { getBEMClasses } from '../helpers/cssClassesHelper';
import '../styles/components/menuComponent.css';

const MenuClasses = getBEMClasses(['menu']);

class MenuComponent extends React.Component {
	state = {
		open: false,
	  };

	handleToggle = () => {
		this.setState({ open: !this.state.open });
	};

	handleClose = () => {
		this.setState({ open: false });
	};

	populateMenuItems =() =>{
		const MenuList =[[scheduleIcon, 'Schedule Consultation'],
		[flagIcon, 'Schedule On-Site'],
		[conferencIcon, 'Conference'],
		[refIcon, 'Referral'],
		[lcfIcon, 'Show Lead Capture']
		]

		return  MenuList.map(i => 
		<MenuItem className={MenuClasses('item')} onClick={this.handleClose}>
		<img src={i[0]} />
		{ i[1] }
		</MenuItem>)		
	}

	render() {
		const { open } = this.state;

		return (
			<div>
				<Button
					className={MenuClasses('icon-postion')}
					buttonRef={node => {
						this.anchorEl = node;
					}}
					aria-owns={open ? 'menu-list-grow' : null}
					aria-haspopup="true"
					onClick={this.handleToggle}
				>
        			{''}
				</Button>
				<div className={MenuClasses('box-postion')}>
					<Popper open={open} anchorEl={this.anchorEl} transition disablePortal>
						{({ TransitionProps, placement }) => (
							<Grow
								{...TransitionProps}
								id="menu-list-grow"
								style={{
									transformOrigin: placement === 'bottom' ? 'left top 25px' : 'left bottom'
								}}
							>
								<Paper>
									<ClickAwayListener onClickAway={this.handleClose}>
										<MenuList className={MenuClasses('box')}>
										{this.populateMenuItems()}
										</MenuList>
									</ClickAwayListener>
								</Paper>
							</Grow>
						)}
					</Popper>
				</div>
			</div>
		);
	}
}

export default MenuComponent;
