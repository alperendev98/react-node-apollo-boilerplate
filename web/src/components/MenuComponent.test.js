import React from 'react';
import { shallow, mount } from 'enzyme';
import { getBEMClasses } from '../helpers/cssClassesHelper';
import MenuComponent from './MenuComponent';

describe('MenuComponent.js TESTS', () => {
	const props = { history: { push: jest.fn() } };
	let wrapper, component;

	beforeEach(() => {
		wrapper = shallow(<MenuComponent />);
		component = mount(<MenuComponent {...props} />, {});
	});

	it('renders MenuComponent without crashing', () => {
		expect(wrapper.length).toEqual(1);
	});

	it('should match previos snapshot for MenuComponent', () => {
		expect(wrapper).toMatchSnapshot();
	});

	it('checks the onClick when menu is clicked', () => {
		const instance = component.instance();
		jest.spyOn(instance, 'handleToggle');

		instance.handleToggle();
		expect(instance.handleToggle).toHaveBeenCalled();
	});

	it('checks close menu on outside click', () => {
		const instance = component.instance();
		jest.spyOn(instance, 'handleClose');

		instance.handleClose();
		expect(instance.handleClose).toHaveBeenCalled();
	});

});
