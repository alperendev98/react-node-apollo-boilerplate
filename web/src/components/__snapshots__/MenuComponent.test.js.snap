// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MenuComponent.js TESTS should match previos snapshot for MenuComponent 1`] = `
<div>
  <WithStyles(Button)
    aria-haspopup="true"
    aria-owns={null}
    buttonRef={[Function]}
    className="menu__icon-postion"
    onClick={[Function]}
  />
  <div
    className="menu__box-postion"
  >
    <WithTheme(Popper)
      disablePortal={true}
      open={false}
      transition={true}
    >
      <Component />
    </WithTheme(Popper)>
  </div>
</div>
`;
